di-ipv-credential-issuer-stub/src/main/java/uk/gov/di/ipv/stub/cred/auth/ClientJwtVerifier.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.di.ipv.stub.cred.auth;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.oauth2.sdk.ParseException;
+import com.nimbusds.oauth2.sdk.auth.PrivateKeyJWT;
+import com.nimbusds.oauth2.sdk.auth.verifier.ClientAuthenticationVerifier;
+import com.nimbusds.oauth2.sdk.auth.verifier.InvalidClientException;
+import com.nimbusds.oauth2.sdk.id.Audience;
+import spark.QueryParamsMap;
+import uk.gov.di.ipv.stub.cred.config.ClientConfig;
+import uk.gov.di.ipv.stub.cred.config.CredentialIssuerConfig;
+import uk.gov.di.ipv.stub.cred.error.ClientAuthenticationException;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ClientJwtVerifier {
+
+    public static final String AUTHENTICATION_METHOD = "authenticationMethod";
+    public static final String NONE = "none";
+
+    private final ClientAuthenticationVerifier<Object> verifier;
+
+    public ClientJwtVerifier() {
+        this.verifier = getPopulatedClientAuthVerifier();
+    }
+
+    public void authenticateClient(QueryParamsMap requestParams)
+            throws ClientAuthenticationException {
+
+        PrivateKeyJWT authenticationJwt;
+        try {
+            authenticationJwt = PrivateKeyJWT.parse(listifyParamValues(requestParams));
+        } catch (ParseException e) {
+            throw new ClientAuthenticationException(e);
+        }
+
+        ClientConfig clientConfig = CredentialIssuerConfig.getClientConfig(authenticationJwt.getClientID().getValue());
+        if (clientConfig == null) {
+            throw new ClientAuthenticationException(String.format("Config for client ID '%s' not found", authenticationJwt.getClientID().getValue()));
+        }
+
+        if (clientConfig.getJwtAuthentication().get(AUTHENTICATION_METHOD).equals(NONE)) {
+            return;
+        }
+
+        try {
+            verifier.verify(authenticationJwt, null, null);
```

Comment:
Do you know if this will test against the expiry ttl time being too far in the future?

Reply:
I don't think it does. I think we agreed that for the stub we're not too concerned about this. I've spoken to @SunilDGDS about having to do it as an extra check in his core-back verifying PR though.

Reply:
Ah okay thats sounds good 👍